x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/authc/ApiKeyIntegTests.java

```diff
@@ -1645,6 +1645,90 @@ public void testBulkUpdateApiKeysWithDuplicates() throws ExecutionException, Int
         assertThat(response.getNoops(), empty());
     }
 
+    public void testBulkUpdateApiKeysWithDifferentLimitedByRoleDescriptorsForSameUser() throws ExecutionException, InterruptedException,
```

Comment:
Sneaking in a test I only thought of while working on the YAML tests. It's quite lengthy and specific so I think it's best placed as in the integration test suite.